python/lsst/pipe/tasks/metrics.py

```diff
@@ -0,0 +1,226 @@
+    pass
+
+
+@register("numDeblendChildSciSources")
+class NumberDeblendChildSourcesMetricTask(MetricTask):
+    """Task that computes the number of science sources created
+    through deblending.
+
+    This task only counts final deblending products; i.e., it does not count
+    sources produced by deblending that were themselves deblended by later
+    processing. If sky source information is present, sky sources are excluded.
+
+    Notes
+    -----
+    The task excludes any non-sky sources in the catalog, but it does
+    not require that the catalog include a ``sky_sources`` column.
+    """
+    _DefaultName = "numDeblendChildSciSources"
+    ConfigClass = NumberDeblendChildSourcesMetricConfig
+
+    def run(self, sources):
+        """Count the number of science sources created by deblending.
+
+        Parameters
+        ----------
+        sources : `lsst.afw.table.SourceCatalog` or `None`
+            A science source catalog, which may be empty or `None`.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            A `~lsst.pipe.base.Struct` containing the following component:
+
+            ``measurement``
+                the total number of science sources from deblending
+                (`lsst.verify.Measurement`). If no deblending information is
+                available in ``sources``, this is `None`.
+
+        Raises
+        ------
+        MetricComputationError
+            Raised if ``sources`` is missing mandatory keys for
+            source catalogs.
+        """
+        if sources is None:
+            self.log.info("Nothing to do: no catalogs found.")
+            meas = None
+        elif "deblend_nChild" not in sources.schema:
+            self.log.info("Nothing to do: no deblending performed.")
+            meas = None
+        else:
+            try:
+                children = ((sources["parent"] != 0)            # deblend child
+                            & (sources["deblend_nChild"] == 0)  # not deblended
+                            )
+                nChildren = np.count_nonzero(_filterSkySources(sources, children))
+            except LookupError as e:
+                # Probably "parent"; all other columns already checked
+                raise MetricComputationError("Invalid input catalog") from e
+            meas = Measurement(self.config.metricName, nChildren * u.dimensionless_unscaled)
+
+        return Struct(measurement=meas)
+
+
+def _filterSkySources(catalog, matches):
+    """Filter out any sky sources from a vector of matches.
+
+    If no sky source information is available, all sources are assumed to
+    be non-sky.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SourceCatalog`
+        The catalog to filter.
+    matches : `numpy.ndarray` [`bool`], (N,)
+        A vector of existing source matches, of the same length as ``catalog``,
+        where matched sources are marked `True`.
```

Comment:
Except that `matches` is filtered by all criteria _except_ sky source. I could say "selection" instead of "matches" (which is now confusing me), but I don't think that addresses your point?

Reply:
Oh right, these are the not-sky-sources. Oy. Maybe something like "matched real sources" (as an opposite of sky sources)? As evidenced by me mixing it up, "matches" is not super illuminating on its own.